### ISSUE TITLE
Troubleshooting mesh networks

### ISSUE THREAD

**Issue by m-anish** [eka-foundation/home#19]
How can we make mesh networks more easy to troubleshoot. We need to think in terms of decentralization and user experience.

**Comment by m-anish** [comment]
Some ideas:

* How to make the mesh nodes communicate whats wrong
* Must resolve to actionable human steps.
* Think of how this may be done in a decentralized way.

**Comment by m-anish** [comment]
Actually it would be really nice to have @sboruah inputs on this, especially the User Interaction part.

**Comment by mikkokotila** [comment]
It seems that *"How to make the mesh nodes communicate whats wrong"* would be the simplest to overcome with `nmap -sn 172.20.10.0/24 -oG - | grep Host | cut -d ' ' -f2` and then comparing that to an expected list. What do you think?

**Comment by m-anish** [comment]
Sure we can think of approaches. I think we can already glean similar info to what you describe above from what SECN shows in its status tab.

There are other factors at play too:

1. What if your antenna is pointing in the wrong direction
2. How is the wifi signal strength - how is the link bandwidth
3. Something's up with the power section
4. Other unforeseeable issues 

And then the UX part. :) 

If possible, would like to have a short call amongst the three of us (@mikkokotila @sboruah me) sometime and I can explain all the factors at play.

**Comment by mikkokotila** [comment]
Great :) Also, I made a mistake ... I think nmap address possibly the part "what is the network to do technically to find out which nodes are down" ... this might be a very simple, yet robust way to achieve that assuming that in a mesh the single node is able to propagate the nmap probe throughout the mesh in the first place. Maybe next level complexity is to capture the signal strength from node to node, number of connections a node has to other nodes, etc. and start going towards a more heuristic direction. I think @energyguy3 will be a great contributor in this topic as well. 

EDIT: to add here, I think within the resources we have available in the mesh, we can already perform shannon entropy based anomaly detection to support this end.

**Comment by mikkokotila** [comment]
The second part is the UX. It seems that as a preliminary step, is that we create a taxonomy of all the network issues that are already known for us.

**Comment by m-anish** [comment]
I agree. Once we sort #22 out, then we can work on the design/UX part. Removing @sboruah for now from this.

**Comment by m-anish** [comment]
Okay, can we schedule a short call between @mikkokotila @EnergyGuy3 @pdey me to start narrowing down on actionable design steps?

**Comment by mikkokotila** [comment]
I think that's a great way to proceed. He should be available on one of the days next week. Maybe better later in the week as he is coming from holiday.

**Comment by m-anish** [comment]
@mikkokotila Cool. Please coordinate with him and suggest a time. My schedule is very flexible, and hopefully so is Prasenjit's.

**Comment by m-anish** [comment]
We need to start moving on this, so I have reassigned this to me in context of short term goals for this season. If you want to discuss longer term design with @energyguy and the rest of us, open a separate issue about it.

**Comment by mikkokotila** [comment]
Yes I agree, this will be the first thing I will work on when I get the NUC (will pick it up from post later today). I will create a simple monitoring service which we can plug into any which way we want. Shall we do a separate thread on that (as it will be more specific on feature set)?

**Comment by m-anish** [comment]
Yes. Lets open another issue.
